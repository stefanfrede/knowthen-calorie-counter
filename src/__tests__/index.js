import flyd from 'flyd';

import initModel from '../model';

it('renders init model correctly', () => {
  const model = () => initModel;

  const update = flyd.stream();
  const models = flyd.scan((model, func) => func(model), model(), update);

  expect(models()).toMatchSnapshot();
});

it('updates init model correctly', () => {
  const model = () => initModel;
  const updatedModel = {
    description: 'Lunch',
    calories: 960,
    showForm: false,
    nextId: 0,
    editId: null,
    meals: [],
  };

  const update = flyd.stream();
  const models = flyd.scan((model, func) => func(model), model(), update);

  models(updatedModel);

  expect(models()).toMatchSnapshot();
});
