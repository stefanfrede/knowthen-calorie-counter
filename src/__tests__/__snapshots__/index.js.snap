// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders init model correctly 1`] = `
Object {
  "calories": 480,
  "description": "Breakfast",
  "editId": null,
  "meals": Array [],
  "nextId": 0,
  "showForm": false,
}
`;

exports[`updates init model correctly 1`] = `
Object {
  "calories": 960,
  "description": "Lunch",
  "editId": null,
  "meals": Array [],
  "nextId": 0,
  "showForm": false,
}
`;
